components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -1075,6 +1078,24 @@ fn cast_int_as_time(
     }
 }
 
+#[rpn_fn(capture = [ctx, extra])]
+fn cast_year_as_time(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra,
+    year: &Int,
+) -> Result<Option<Time>> {
+    let year = *year;
```

Comment:
```
let year = *year as u32
```

So that we can reduce the `MIN_YEAR.into()` and `year as u32` later?

Reply:
`*year as u32` will truncate the value of i64. It may accidentally turn a value from invalid to valid.

Reply:
I got it.